internal/proxy/impl.go

```diff
@@ -1697,6 +1697,138 @@ func (node *Proxy) ShowPartitions(ctx context.Context, request *milvuspb.ShowPar
+	IDs2Names := make(map[int64]string)
+	partitionIDs := make([]int64, 0)
+	for _, partitionName := range request.PartitionNames {
+		partitionID, err := globalMetaCache.GetPartitionID(ctx, request.CollectionName, partitionName)
+		if err != nil {
+			return 0, err
+		}
+		IDs2Names[partitionID] = partitionName
+		partitionIDs = append(partitionIDs, partitionID)
+	}
+	resp, err := node.queryCoord.ShowPartitions(ctx, &querypb.ShowPartitionsRequest{
+		Base: &commonpb.MsgBase{
+			MsgType:   commonpb.MsgType_ShowPartitions,
+			MsgID:     request.Base.MsgID,
+			Timestamp: request.Base.Timestamp,
+			SourceID:  request.Base.SourceID,
+		},
+		CollectionID: collectionID,
+		PartitionIDs: partitionIDs,
+	})
+	if err != nil {
+		return 0, err
+	}
+	if len(resp.InMemoryPercentages) != len(partitionIDs) {
+		return 0, errors.New("fail to show partitions from the querycoord, invalid data num")
+	}
+	var progress int64
+	for _, p := range resp.InMemoryPercentages {
+		progress += p
+	}
+	progress /= int64(len(partitionIDs))
+	return progress, nil
+}
+
+func (node *Proxy) GetLoadingProgress(ctx context.Context, request *milvuspb.GetLoadingProgressRequest) (*milvuspb.GetLoadingProgressResponse, error) {
+	if !node.checkHealthy() {
+		return &milvuspb.GetLoadingProgressResponse{Status: unhealthyStatus()}, nil
+	}
+	method := "GetLoadingProgress"
+	tr := timerecord.NewTimeRecorder(method)
+	log.Debug(method, zap.String("collection_name", request.CollectionName),
+		zap.Strings("partition_name", request.PartitionNames))
+
+	getErrResponse := func(err error) *milvuspb.GetLoadingProgressResponse {
+		logger.Error("fail to get loading progress", zap.String("collection_name", request.CollectionName),
+			zap.Strings("partition_name", request.PartitionNames), zap.Error(err))
+		return &milvuspb.GetLoadingProgressResponse{
+			Status: &commonpb.Status{
+				ErrorCode: commonpb.ErrorCode_UnexpectedError,
+				Reason:    err.Error(),
+			},
+		}
+	}
+	if err := validateCollectionName(request.CollectionName); err != nil {
+		return getErrResponse(err), nil
+	}
+	collectionID, err := globalMetaCache.GetCollectionID(ctx, request.CollectionName)
+	if err != nil {
+		return getErrResponse(err), nil
+	}
+	msgBase, err := node.getMsgBase()
+	if err != nil {
+		return getErrResponse(err), nil
+	}
+	if request.Base == nil {
+		request.Base = msgBase
+	} else {
+		request.Base.MsgID = msgBase.MsgID
+		request.Base.Timestamp = msgBase.Timestamp
+		request.Base.SourceID = msgBase.SourceID
+	}
+
+	var progress int64
+	if len(request.GetPartitionNames()) == 0 {
+		if progress, err = node.getCollectionProgress(ctx, request, collectionID); err != nil {
+			return getErrResponse(err), nil
+		}
+	} else {
+		if progress, err = node.getPartitionProgress(ctx, request, collectionID); err != nil {
+			return getErrResponse(err), nil
+		}
+	}
+
+	logger.Debug(method+"success", zap.String("collection_name", request.CollectionName),
```

Comment:
log.Info(
		rpcDone(method),
		zap.String("traceID", traceID),
		zap.String("role", typeutil.ProxyRole),
		zap.Int64("MsgID", cit.ID()),
		zap.Uint64("BeginTs", cit.BeginTs()),
		zap.Uint64("EndTs", cit.EndTs()),
		zap.String("db", request.DbName),
		zap.String("collection", request.CollectionName),
		zap.String("field", request.FieldName),
		zap.Any("extra_params", request.ExtraParams))

also print the percentage int the log out.